packages/portalnetwork/src/wire/utp/Socket/UtpSocket.ts

```diff
@@ -87,213 +116,196 @@ export class UtpSocket extends EventEmitter {
     }
-    this.logger(`${PacketType[type]}   sent - ${messageData[type]}`)
-    this.utp.emit('Send', this.remoteAddress, msg, ProtocolId.HistoryNetwork, true)
+    this.logger(`${PacketType[packet.header.pType]}   sent - ${messageData[type]}`)
+    this.emit('send', this.remoteAddress, msg, ProtocolId.HistoryNetwork, true)
     return msg
   }
 
-  async sendSynPacket(packet: Packet): Promise<ConnectionState> {
-    await this.sendPacket(packet, PacketType.ST_SYN)
-    this.state = ConnectionState.SynSent
-    return this.state
-  }
-
-  async sendSynAckPacket(packet: Packet): Promise<void> {
-    await this.sendPacket(packet, PacketType.ST_STATE)
+  createPacket<T extends PacketType>(
+    opts: ICreatePacketOpts<T> = {} as ICreatePacketOpts<T>
+  ): Packet<T> {
+    opts.pType === PacketType.ST_DATA && this.seqNr++
+    const extension = 'bitmask' in opts ? 1 : 0
+    const params: ICreate<T> = {
+      ...opts,
+      seqNr: this.seqNr,
+      ackNr: this.ackNr,
+      extension,
+    }
+    return this.packetManager.createPacket<T>(params)
   }
 
-  async sendDataPacket(packet: Packet): Promise<Packet> {
-    this.state = ConnectionState.Connected
-    this.outBuffer.set(packet.header.seqNr, packet.header.timestampMicroseconds)
-    this.cur_window = this.outBuffer.size * DEFAULT_PACKET_SIZE
-    await this.sendPacket(packet, PacketType.ST_DATA)
-    this.seqNr++
-    return packet
+  async sendSynPacket(): Promise<void> {
+    const p = this.createPacket({ pType: PacketType.ST_SYN })
+    await this.sendPacket(p, PacketType.ST_SYN)
+    this.state = ConnectionState.SynSent
   }
-
-  async sendStatePacket(packet: Packet): Promise<void> {
+  async sendAckPacket(bitmask?: Uint8Array): Promise<void> {
+    const packet = bitmask
+      ? this.createPacket({ pType: PacketType.ST_STATE, bitmask })
+      : this.createPacket({ pType: PacketType.ST_STATE })
     await this.sendPacket(packet, PacketType.ST_STATE)
   }
-
-  async sendResetPacket(packet: Packet) {
+  async sendSynAckPacket(): Promise<void> {
+    this.state = ConnectionState.SynRecv
+    this.logger(`Connection State: SynRecv`)
+    await this.sendAckPacket()
+  }
+  async sendResetPacket() {
     this.state = ConnectionState.Reset
-    await this.sendPacket(packet, PacketType.ST_RESET)
+    const packet = this.createPacket<PacketType.ST_RESET>({ pType: PacketType.ST_RESET })
+    await this.sendPacket<PacketType.ST_RESET>(packet, PacketType.ST_RESET)
   }
-
-  async sendFinPacket(packet: Packet): Promise<Buffer> {
+  async sendFinPacket(): Promise<void> {
+    const packet = this.createPacket<PacketType.ST_FIN>({ pType: PacketType.ST_FIN })
     this.finNr = packet.header.seqNr
-    return await this.sendPacket(packet, PacketType.ST_FIN)
+    await this.sendPacket<PacketType.ST_FIN>(packet, PacketType.ST_FIN)
+  }
+  async sendDataPacket(bytes: Uint8Array): Promise<void> {
+    this.state = ConnectionState.Connected
+    await this.packetManager.congestionControl.canSend()
```

Comment:
Will this ever return false?  If not, can we skip the boolean and just await it?